### PR TITLE
#553/fix buttons and remove pagination from get levels endpoints

### PR DESCRIPTION
**Pull request by Damrod**
as explained #553

### CODE REVIEW

File: transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts [hunk 1]
```diff
@@ -94,20 +93,20 @@ export class UserLevelPostgresRepository
...
-                 OR ${gameKeys.PLAYERTWOUSERNAME} = $1;`,
+                  OR ${gameKeys.PLAYERTWOUSERNAME} = $1;`,
         values: [username],
       });
     }
     return gamesData ? gamesData.map((game) => new Game(game)) : null;
   }
 
-  async getWinGameRatio(
+  async getGameResults(
```
Comment (gbudau) [inline transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts:103]: suggestion: I think you already wrote this in another PR. We could do this with SQL with a query like:

```ts
  async getGameResults(
    username: string,
    gameMode?: GameMode,
  ): Promise<GameStats | null> {
    let gamesStatsData: GameStatsData[] | null = null;
    if (gameMode) {
      gamesStatsData = await makeQuery<GameStatsData>(this.pool, {
        text: `
SELECT
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} > ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} > ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS wins,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} < ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} < ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS losses,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1) AND ${gameKeys.PLAYERONESCORE} = ${gameKeys.PLAYERTWOSCORE} THEN 1 END) AS draws
FROM ${table.GAME}
WHERE (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1)
AND ${gameKeys.GAMEMODE} = $2;
        `,
        values: [username, gameMode],
      });
    } else {
      gamesStatsData = await makeQuery<GameStatsData>(this.pool, {
        text: `
SELECT
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} > ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} > ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS wins,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} < ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} < ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS losses,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1) AND ${gameKeys.PLAYERONESCORE} = ${gameKeys.PLAYERTWOSCORE} THEN 1 END) AS draws
FROM ${table.GAME}
WHERE (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1);
        `,
        values: [username],
      });
    }
    return gamesStatsData && gamesStatsData.length
      ? new GameStats(gamesStatsData[0])
      : null;
  }
```

**Review by gbudau — APPROVED** [review]
Thank you for these improvements! LGTM :tada:

**Review by fizcamposancos — APPROVED** [review]
LGTM! Thanks

File: transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts (hunk 1 above)
Comment (Damrod) [reply transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts:103]: This looks very nice. Will do.

Comment (Damrod) [reply transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts]: done. Thank you!

**Pull request merged**